tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -421,6 +421,45 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system",
 	},
 
+	/** Firmware layout for the C59v2 */
+	{
+		.id	= "ARCHER-C59-V2",
+		.vendor	= "",
+		.support_list =
+			"SupportList:\r\n"
+			"{product_name:Archer C59,product_ver:2.0.0,special_id:00000000}\r\n"
+			"{product_name:Archer C59,product_ver:2.0.0,special_id:55530000}\r\n",
+		.support_trail = '\x00',
+		.soft_ver = "soft_ver:2.0.0 Build 20161206 rel.7303\n",
+
+		.partitions = {
+			{"factory-boot", 0x00000, 0x20000},
+			{"fs-uboot", 0x20000, 0x10000},
+			{"default-mac", 0x30000, 0x00200},
+			{"pin", 0x30200, 0x00200},
+			{"device-id", 0x30400, 0x00100},
+			{"product-info", 0x30500, 0x0fb00},
+			{"os-image", 0x40000, 0x180000},
```

Comment:
The mentioned patch is now merged to master. Please apply the same changes as in 2f23a0583a3092d9076ceecca16be0f8ccdce217 and 97da92b1277a40fa7daf354ed52b8ea25e06a779. Please take care to remove the kernel and rootfs partitions from the MTDPARTS.